packages/zowe-explorer-api/__tests__/__unit__/profiles/ProfilesCache.unit.test.ts

```diff
@@ -153,6 +164,20 @@ describe("ProfilesCache", () => {
         expect(Object.keys(keyring).length).toBe(5);
     });
 
+    it("addToConfigArray should set the profileTypeConfigurations array", () => {
+        const profCache = new ProfilesCache(fakeLogger as unknown as zowe.imperative.Logger);
+        profilemetadata.push(profilemetadata[0]);
+        profCache.addToConfigArray(profilemetadata);
+        expect(profCache.profileTypeConfigurations).toEqual(profilemetadata.filter((a, index) => index == 0));
```

Comment:
clever way to test that the `addToConfigArray` method is intended to avoid duplicates 😋